proposed/2022/DotnetListPackageMachineReadableJsonOutput.md

```diff
@@ -772,6 +772,173 @@ Outputs json for format version 1, if it's not specified then latest version'll
+              "resolvedVersion": "4.8.0-preview3.5278"
+            },
+            {
+              "id": "Text2Xml.Lib",
+              "requestedVersion": "[1.1.2, 2.0.0)",
+              "resolvedVersion": "1.1.2"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### `> dotnet list NonExisting.csproj package`
+
+`NonExisting.csproj` doesn't exist or wrong path.
+
+```dotnetcli
+Could not find file or directory 'C:\Users\userA\repos\MainApp\src\lib\NonExisting.csproj'.
+```
+
+#### `> dotnet list package --format json`
+
+```json
+{
+  "version": 1,
+  "parameters": "",
+  "problems": [
+    {
+        "project": "src/lib/NonExisting.csproj",
+        "message": "Could not find file or directory 'C:/Users/userA/repos/MainApp/src/lib/NonExisting.csproj'"
+    }
+  ]
+}
+```
+
+#### `> dotnet list package`
+
+`MyProjectB` is package.config type project, so it's not supported.
+
+```dotnetcli
+Project 'MyProjectA' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package                      Requested             Resolved
+   > Microsoft.Extensions.Primitives      [1.0.0, 5.0.0]        1.0.0
+   > NuGet.Commands                       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib                         [1.1.2, 2.0.0)        1.1.2
+
+The project `C:\Users\userA\repos\MainApp\src\lib\MyProjectB.csproj` uses package.config for NuGet packages, while the command works only with package reference projects.
+```
+
+#### `> dotnet list package --format json`
+
+```json
+{
+  "version": 1,
+  "parameters": "",
+  "problems": [
+    {
+        "project": "src/lib/MyProjectB.csproj",
+        "message": "The project `C:/Users/userA/repos/MainApp/src/lib/MyProjectB.csproj`` uses package.config for NuGet packages, while the command works only with package reference projects."
+    }
+  ],
+  "projects": [
+    {
+      "path": "src/lib/MyProjectA.csproj",
+      "frameworks": [
+        {
+          "framework": "netcoreapp3.1",
+          "topLevelPackages": [
+            {
+              "id": "Microsoft.Extensions.Primitives",
+              "requestedVersion": "[1.0.0, 5.0.0]",
+              "resolvedVersion": "1.0.0"
+            },
+            {
+              "id": "NuGet.Commands",
+              "requestedVersion": "4.8.0-preview3.5278",
+              "resolvedVersion": "4.8.0-preview3.5278"
+            },
+            {
+              "id": "Text2Xml.Lib",
+              "requestedVersion": "[1.1.2, 2.0.0)",
+              "resolvedVersion": "1.1.2"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### `> dotnet list package --format json --output-version 3`
+
+Since output version `3` is not available, then it'll default to latest available version, and log about unsupported format version request into json result.
+
+```json
+{
+  "version": 1,
+  "parameters": "",
+  "problems": [
+    {
+        "project": "",
```

Comment:
For general error, should I skip `project` property or not? Maybe put place holder `*` for value?

Reply:
I think not including the project property when empty is ok.

Reply:
Ok. Addressed, not including empty property.